app/src/main/java/com/fastlink/zekrapp/viewModel/ZekirViewModel.kt

```diff
@@ -0,0 +1,50 @@
+package com.fastlink.zekrapp.viewModel
+
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.lifecycle.ViewModel
+import com.fastlink.zekrapp.appData.ZekirSingleton
+import com.fastlink.zekrapp.appData.model.ZekirModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class ZekirViewModel(categoryId: Int) : ViewModel() {
+    val zekirCounter = mutableIntStateOf(0)
+    var zekirs: List<ZekirModel> = emptyList()
+
+    init {
+        getZekirByCategoryId(categoryId)
+    }
+
+    private fun getZekirByCategoryId(categoryId: Int) {
+        resetZekirCounter()
```

Comment:
the function name is called getZekir... but here you are doing something else which is resetting. Move this to the init block. Because your function name should be what the function is doing.

Reply:
i didn't notate this